calyx/src/ir/primitives.rs

```diff
@@ -84,30 +78,46 @@ impl Primitive {
 
 /// Definition of a port.
 #[derive(Clone, Debug)]
-pub struct PortDef {
+pub struct PortDef<W> {
     /// The name of the port.
     pub name: Id,
-    /// The width of the port. Can be either a number ([`Width::Const`]) or
-    /// a parameter ([`Width::Param`]).
```

Comment:
Maybe preserving some of this documentation intent somewhere would help readability in posterity?